Make the browser example watch independent from start

hexa00 opened the pull request:
This patch makes it so that npm start no longer start watchers.

So that the following developement workflow is now possible:

#Watch the backend, frontend and start the file-depencency-updater 
npm run watch 

#Run the backend and frontend
npm start

Signed-off-by: Antoine Tremblay <antoine.tremblay@ericsson.com>

akosyakov commented on examples/browser/package.json:
```diff
@@ -21,11 +21,12 @@
     "start:frontend": "webpack-dev-server --open",
     "cold:start": "npm run clean && npm run build && npm start",
-    "watch:backend": "tsc --watch",
+    "watch:backend": "tsc --watch -p ../../",
```
Why ` -p ../../`?

hexa00 replied:
Yes this is wrong, my intention was to watch theia-core, I'll add a new watch:core with tsc --watch -p ../../ and leave the watch:backend as it was

akosyakov replied:
why not run `watch` on `theia-core` itself?

hexa00 replied:
Well it's another shell & another command it's running the file-updater anyway so like this all you would have to do is npm run watch and all would be taken care of ?

akosyakov replied:
ok, make sense

hexa00 commented:
With this however it makes it very apparent that the file-updater is not working properly.

We get into the same kind of errors we had with chokidar like :

```
[watch-frontend] SyntaxError: Unexpected end of JSON input
[watch-frontend]     at JSON.parse (<anonymous>)
[watch-frontend]     at /home/eantotr/ownCloud/work/theia/examples/browser/node_modules/source-map-loader/index.js:44:22
[watch-frontend]     at tryToString (fs.js:449:3)
[watch-frontend]     at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:436:12)
[watch-frontend] 
ch-backend] node_modules/theia-core/lib/application/browser/application.d.ts(3,33): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/command.d.ts' is not a module.
[watch-backend] node_modules/theia-core/lib/application/browser/application.d.ts(4,36): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/keybinding.d.ts' is not a module.
[watch-backend] node_modules/theia-core/lib/application/browser/menu/menu-plugin.d.ts(3,33): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/command.d.ts' is not a module.
[watch-backend] node_modules/theia-core/lib/application/browser/menu/menu-plugin.d.ts(4,36): error TS2306: File '/home/eantotr/ownCloud/work/theia/examples/browser/node_modules/theia-core/lib/application/common/keybinding.d.ts' is not a module.

```
This makes it unusable for me :(

I think we should accept this pull request and work on the file-updater separately

akosyakov commented:
@hexa00 Will you rebase or close it?

hexa00 commented:
Yes I just rebased and will merge but I removed the localdep watch and the core watch from there since it was just not working.

So watch only watches the frontend/backend and the dev is responsible to watch the rest elsewhere or run a build...

Actually running a build I think should be the preferred way, I'm still experimenting before writing something I'm confident about in the dev doc.
